plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/PreferenceConstants.java

```diff
@@ -40,4 +52,9 @@ static IPreferenceStore getPreferenceStore() {
   static IEclipsePreferences getPreferenceNode() {
     return InstanceScope.INSTANCE.getNode(BUNDLEID);
   }
+
+  static void initializeDefaults() {
```

Comment:
Hmm… I think we could actually reduce this to just using the CLOUD_SDK_PATH preference.  We use the Managed SDK only if a Cloud SDK was not found in the usual places.  I'm thinking that if I launch Eclipse with `GOOGLE_CLOUD_SDK_HOME` set to a location, it should override any managed location.

Reply:
I think it's good to have an explicit preference for the management option, as there are cases where the behavior becomes unclear. For example, suppose CT4E did not find the SDK in the usual places so it has been set to auto-managed. What if the user later installs an SDK manually? Should CT4E continue to auto-manage its SDK or switch to use the local one? Similarly, what if the user was using a local SDK and then the SDK became temporarily available because the user moved the directory? Should CT4E start downloading and installing its own SDK?